lang/LangSource/SC_LanguageConfig.cpp

```diff
@@ -93,47 +103,51 @@ bool SC_LanguageConfig::removeIncludedDirectory(const Path& path) { return remov
 
 bool SC_LanguageConfig::removeExcludedDirectory(const Path& path) { return removePath(mExcludedDirectories, path); }
 
+static void processPathList(const char* nodeName, YAML::Node& doc,
+                            const std::function<void(const boost::filesystem::path&)>& func) {
+    const YAML::Node& items = doc[nodeName];
+    if (items && items.IsSequence()) {
+        for (auto const& item : items) {
+            const std::string& path = item.as<std::string>("");
+            if (!path.empty()) {
+                const boost::filesystem::path& native_path = SC_Codecvt::utf8_str_to_path(path);
+                func(native_path);
+            }
+        }
+    }
+}
+
+static void processBool(const char* nodeName, YAML::Node& doc, const std::function<void(bool)>& successFunc,
+                        const std::function<void(void)>& failFunc) {
+    const YAML::Node& item = doc[nodeName];
+    if (item) {
```

Comment:
Library doesn't compile if 'excludeDefaultPaths' is not present in YAML. Could be fixed like this:
```diff
--- a/lang/LangSource/SC_LanguageConfig.cpp
+++ b/lang/LangSource/SC_LanguageConfig.cpp
@@ -123,11 +123,12 @@ static void processBool(const char* nodeName, YAML::Node& doc, const std::functi
     if (item) {
         try {
             successFunc(item.as<bool>());
+            return;
         } catch (...) {
             postfl("Warning: Cannot parse config file entry \"%s\"\n", nodeName);
-            failFunc();
         }
-    }
+    }
+    failFunc(); // call failFunc if item is not found
 }
```